source/util/convert_to_iso_string.ts

```diff
@@ -0,0 +1,20 @@
+function isIsoDate(str: string) {
+  if (!/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(str)) return false;
+  const d = new Date(str);
+  return d instanceof Date && !isNaN(d.getTime()) && d.toISOString() === str; // valid date
+}
+
+export function convertToISOString(data: string | Date) {
+  if (
+    data &&
+    // if this is a date object of type moment or dayjs, or regular date object (all of them has toISOString)
+    ((typeof data !== "string" && typeof data.toISOString === "function") ||
+      // if it's a ISO string already
+      (typeof data == "string" && isIsoDate(data)))
```

Comment:
Why do we encode already formatted strings?

Reply:
to make sure they are in UTC format.

Reply:
Just checking, do we have a use case to supporting these? It seems like you have to jump through a few hoops to format dates like this.

Reply:
Nothing that I'm aware of, I thought it was better to support encoding all ISO8601 compatible strings than just a subset of them.